File: Benchmarking/src/gen_1115_37.scala
import java.lang.System.nanoTime
import scala.collection.immutable.Vector
object Gen extends App {
  case class s0 (
    p0: Vector[Vector[Double]],
    p1: Vector[Vector[Double]]
  )
  case class s1 (
    p0: Vector[s0],
    p1: s0
  )
  @noinline
  def f0(v0: Vector[Vector[s1]], v1: s0, v2: Double): Double = {
    val v7: Vector[s1] = v0(0)
    val v4: s1 = v7(0)
    val v12: Vector[s0] = v4.p0
    val v20: s1 = s1(v12, v1)
    val v29: s0 = v20.p1
    val v16: Vector[Vector[Double]] = v29.p1
    val v19: Vector[Double] = v16(2)
    val v71: Vector[Double] = v16(0)
    val v75: Vector[Vector[Double]] = Vector(v71, v19)
    val v82: Vector[Double] = v75(0)
    val v105: Double = v82(0)
    v105
  }
  def benchmark(): Unit = {
    val v0: Vector[Vector[s1]] = Vector(Vector(s1(Vector(s0(Vector(Vector(0.0)), Vector(Vector(1.0), Vector(2.0), Vector(3.0))), s0(Vector(Vector(4.0)), Vector(Vector(5.0), Vector(6.0), Vector(7.0)))), s0(Vector(Vector(8.0)), Vector(Vector(9.0), Vector(10.0), Vector(11.0))))), Vector(s1(Vector(s0(Vector(Vector(12.0)), Vector(Vector(13.0), Vector(14.0), Vector(15.0))), s0(Vector(Vector(16.0)), Vector(Vector(17.0), Vector(18.0), Vector(19.0)))), s0(Vector(Vector(20.0)), Vector(Vector(21.0), Vector(22.0), Vector(23.0))))), Vector(s1(Vector(s0(Vector(Vector(24.0)), Vector(Vector(25.0), Vector(26.0), Vector(27.0))), s0(Vector(Vector(28.0)), Vector(Vector(29.0), Vector(30.0), Vector(31.0)))), s0(Vector(Vector(32.0)), Vector(Vector(33.0), Vector(34.0), Vector(35.0))))))
    val v1: s0 = s0(Vector(Vector(36.0)), Vector(Vector(37.0), Vector(38.0), Vector(39.0)))
    val v2: Double = 40.0
    val start = nanoTime()
    var result: Double = 40.0
    (1 to 1000).foreach { _ =>
      result = f0(v0, v1, v2)
    }
    val end = nanoTime()
    println(result)
    println(end - start)
  }
  benchmark()
}

File: Benchmarking/src/gen_1115_37.swift
  import Dispatch
  struct s0 {
    var p0: [[Double]]
    var p1: [[Double]]
  }
  struct s1 {
    var p0: [s0]
    var p1: s0
  }
  @inline(never)
  func f0(_ v0: [[s1]], _ v1: s0, _ v2: Double) -> Double {
    let v7: [s1] = v0[0]
    let v4: s1 = v7[0]
    let v12: [s0] = v4.p0
    let v20: s1 = s1(p0: v12, p1: v1)
    let v29: s0 = v20.p1
    let v16: [[Double]] = v29.p1
    let v19: [Double] = v16[2]
    let v71: [Double] = v16[0]
    let v75: [[Double]] = [v71, v19]
    let v82: [Double] = v75[0]
    let v105: Double = v82[0]
    return v105
  }
  func benchmark() {
    let v0: [[s1]] = [[s1(p0: [s0(p0: [[0.0]], p1: [[1.0], [2.0], [3.0]]), s0(p0: [[4.0]], p1: [[5.0], [6.0], [7.0]])], p1: s0(p0: [[8.0]], p1: [[9.0], [10.0], [11.0]]))], [s1(p0: [s0(p0: [[12.0]], p1: [[13.0], [14.0], [15.0]]), s0(p0: [[16.0]], p1: [[17.0], [18.0], [19.0]])], p1: s0(p0: [[20.0]], p1: [[21.0], [22.0], [23.0]]))], [s1(p0: [s0(p0: [[24.0]], p1: [[25.0], [26.0], [27.0]]), s0(p0: [[28.0]], p1: [[29.0], [30.0], [31.0]])], p1: s0(p0: [[32.0]], p1: [[33.0], [34.0], [35.0]]))]]
    let v1: s0 = s0(p0: [[36.0]], p1: [[37.0], [38.0], [39.0]])
    let v2: Double = 40.0
    let start = DispatchTime.now().uptimeNanoseconds
    var result: Double = 40.0
    for _ in 1...1000 {
      result = f0(v0, v1, v2)
    }
    let end = DispatchTime.now().uptimeNanoseconds
    print(result)
    print(end - start)
  }
  benchmark()
